tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/IncrementalRenaming.scala

```diff
@@ -173,11 +180,11 @@ class IncrementalRenaming @Inject() (tracker: TransformationTracker) extends Tla
       val newName = getNextUniqueFromBase(name)
       val newRenamed = alreadyRenamed + (name -> newName)
       val newArgs = otherArgs.map(rename(newRenamed))
-      val trackedName = tracker.hold(nex, NameEx(newName))
-      OperEx(op, trackedName +: newArgs: _*)
+      val trackedName = tracker.hold(nex, NameEx(newName)(nex.typeTag))
+      OperEx(op, trackedName +: newArgs: _*)(opex.typeTag)
```

Comment:
rather than adding transferring over the implicit type parameter in every branch, might it help with maintainability and readability to apply a function like `preserveType(old: TlaEx, new: TlaEx): TlaEx` to the result of the case analysis? (Perhaps also defining `tracker.hold` to preserve the type (or a variant that will preserve the type)?

Potential benefits:

- The reader can see immediately from the code that this happens in all branches.
- No risk of forgetting to transfer the type when we add another case in the future.
- Slight deduplication of code.
- Done need to add match captures (or what does Scala call `@`?)

Reply:
Yep. I overdid it with implicits a bit. Sure, it makes sense to copy types in many transformations.